target/cheri-common/cheri-translate-utils.h

```diff
@@ -88,30 +90,43 @@ static inline bool have_cheri_tb_flags(DisasContext *ctx, uint32_t flags)
     return (ctx->base.cheri_flags & flags) == flags;
 }
 
-static inline void _generate_ddc_checked_ptr(
+// Checks an address against PCC or DDC.
+// Permissions are checked at translate time
+// Bounds checks can be skipped if CHERI flags indicate bounds would make checks
+// trivial.
+static inline void _generate_special_checked_ptr(
     DisasContext *ctx, CheriTbFlags tb_perm_flags, CheriPermissions req_perms,
-    TCGv_cap_checked_ptr checked_addr, TCGv ddc_offset, target_ulong num_bytes)
+    TCGv_cap_checked_ptr checked_addr, TCGv ddc_offset, target_ulong num_bytes,
+    bool use_ddc)
 {
+    CheriTbFlags full_as_flags =
+        use_ddc ? TB_FLAG_CHERI_DDC_FULL_AS : TB_FLAG_CHERI_PCC_FULL_AS;
+
     if (unlikely(!have_cheri_tb_flags(ctx, tb_perm_flags))) {
-        // DDC is untagged, sealed, or missing PERM_STORE
+        // PCC/DDC is untagged, sealed, or missing PERM_STORE
         TCGv_i32 tperms = tcg_const_i32(req_perms);
         cheri_tcg_prepare_for_unconditional_exception(&ctx->base);
         gen_helper_raise_exception_ddc_perms(cpu_env, tperms);
         tcg_temp_free_i32(tperms);
         return;
     }
-    const bool need_interposition =
-        !have_cheri_tb_flags(ctx, TB_FLAG_CHERI_DDC_NO_INTERPOSE);
+
+    // pcc interpostion currently done mostly by the caller
```

Comment:
Whilst you're touching this function for the comment below please uppercase PCC (wouldn't nit pick it otherwise :P)

Reply:
Well strictly speaking @LawrenceEsswood touched the function :P